### DISCUSSION TITLE
How to trigger multiple actions with a signalSlice?

### DISCUSSION THREAD

**Discussion by kisamoto** [nartc/ngxtension-platform#433] [Q&A] [answered]
I'm using `signalSlice` to manage state within a service.

My use case is managing user preferences. When a user modifies their preferences I'd like it to immediately update the state (so it's reflected in the UI) as well as making a HTTP request to update the preferences in the backend.

As the HTTP request returns the latest saved state I'd then like to re-update the local state with what's returned in the response.

Currently using a `signalState` action source I have to return a partial state for it to update the state. What is the recommended way to:

- Update the local state immediately
- Make a call to the HTTP API
- Update the local state again with the response from the backend

Thanks.

**Comment by eneajaho** [comment]
cc @joshuamorony

**Comment by joshuamorony** [comment, marked as answer]
@kisamoto you should be able to do this by having your `actionSource` switchMap to the http request to the backend (and also map the response to the appropriate partial of the state) and then to satisfy the other requirement to immediately update the local state you can pipe on `startWith` to the http request and have it emit the partial you want to set immediately

**Reply by kisamoto** [reply]
Love this idea - thank you! I'll try it out.
